src/main/scala/org/fortysevendeg/sparksftp/common/SparkUtils.scala

```diff
@@ -41,20 +40,38 @@ object SparkUtils {
       .csv(sourceUri.toString)
   }
 
-  def persistDataFrame(sparkSession: SparkSession, df: DataFrame, name: String) = {
+  def persistDataFrame(
+      sparkSession: SparkSession,
+      df: DataFrame,
+      name: String,
+      partitionBy: Seq[String] = Seq.empty
+  ) = {
     // Persist the dataframes into Hive tables with parquet file format, the default compression for parquet is snappy, that is splittable for parquet.
+    // Another option: externalTable (HDFS, Hive)
     // If we wanted to debug any issue with the databases, we could use this: sparkSession.sparkContext.setLogLevel("DEBUG")
     sparkSession.sql(s"DROP TABLE IF EXISTS ${name}")
-    df.write.mode(SaveMode.Overwrite).format("parquet").saveAsTable(name)
+
+    df.write
+      .mode(SaveMode.Overwrite)
+      .partitionBy(partitionBy: _*)
+      .format("parquet")
+      .saveAsTable(name)
   }
 
   def dataframeToCompressedCsv(df: DataFrame, path: String) = {
-    df.write
+    df.coalesce(1)
```

Comment:
We are repartition everything to just 1 partition, what's the purpose of this? It seems like this is going to be a non-performance operation

Reply:
By having an external filesystem like SFTP, when we added the option of a more complex case, this is a partitioned parquet table in hive, when exporting the CSV, this generated the temporary files on the SFTP file system, each one, deleting the temporary files (attempts) of each other. this is to generate a single CSV to be exported. We could persist each file independently before exporting them, but that would be also more costly than this.